hw2/homework/laughter_prediction/predictors.py

```diff
@@ -86,115 +84,133 @@ def __init__(self, threshold=0.045985743):
 
 
 class RnnPredictor(Predictor):
-    def __init__(self, model_path=None):
+    def __init__(self, model_path="rnn_model"):
         self.model_path = model_path
         self.model = None
-        if model_path is not None:
-            self.model = torch.load(self.model_path)
 
     def save(self):
         torch.save(self.model.state_dict(), self.model_path)
 
-    def train(self, audios, all_tags, batch_size=2, epoches=30):
+    def train(self, audios_mfcc, audios_fbank, all_tags,
+              audios_mfcc_test, audios_fbank_test, all_tags_test,
+              batch_size=2, epoches=30, hidden_size=10):
         """
         :param audios: np.array shape=[n_audios, m_frames, k_features]
                         m_frames - individual for each audio, k_features - same for all audios and frames
         :param all_tags: np.array shape=[n_audios, m_frames, k_classes]
                         m_frames - individual for each audio, k_classes - same for all audios and frames
         :param epoches: int
         """
-        n_audios, audio_length, k_features = audios.shape
-        n_audios, audio_length, k_classes = all_tags.shape
+        _, _, mfcc_k_features = audios_mfcc.shape
+        _, _, fbank_k_features = audios_fbank.shape
+        n_audios, audio_length = all_tags.shape
 
-        audios = torch.tensor(audios).to(device)
+        audios_mfcc = torch.tensor(audios_mfcc).to(device)
+        audios_fbank = torch.tensor(audios_fbank).to(device)
         all_tags = torch.tensor(all_tags).to(device)
+
+        audios_mfcc_test = torch.tensor(audios_mfcc_test).to(device)
+        audios_fbank_test = torch.tensor(audios_fbank_test).to(device)
+        all_tags_test = torch.tensor(all_tags_test).to(device)
+
         if self.model is None:
-            self.model = GRUTagger(n_features=audios.shape[-1], hidden_size=10,
-                                   n_classes=2).to(device)
+            self.model = GRUTagger(n_features_mfcc=mfcc_k_features, n_features_fbank=fbank_k_features,
+                                   hidden_size=hidden_size).to(device)
+        if self.model_path and os.path.isfile(self.model_path):
```

Comment:
In spare time take a look at `pathlib.Path` in my tase creating `Path` object is more pleasant than manually doing file existence checks